### ISSUE TITLE
Update stack by pulling new images

### ISSUE THREAD

**Issue by Dzhuneyt** [docker/swarm#2938]
Running `docker stack deploy` is okay and makes it really easy to deploy a brand new stack of containers.

However, once a stack is running, upgrading all images to their newest versions and updating the containers is really cumbersome.

I think Docker Stack deserves a `docker stack redeploy [existing_stack_name]` command.

**Comment by jony4** [comment]
docker service update --image redis:latest service_name

**Comment by jony4** [comment]
docker service rm service_name 
docker stack deploy service_name_new

Do you mean this?

**Comment by Dzhuneyt** [comment]
No, I mean a built-in way to redeploy a stack (collection of services) after updates to the compose file, with minimum to no downtime.

Currently, it's only possible through trickery and magic (e.g. get all current services, force update them, etc). I can't be the only one to think that not only services are versioned and need upgrading, but also the stack itself (occasionally), right?

**Comment by jony4** [comment]
3. Re-run the docker stack deploy command on the manager, and whatever services need updating are updated:

```sh
$ docker stack deploy -c docker-compose.yml getstartedlab
Updating service getstartedlab_web (id: angi1bf5e4to03qu9f93trnxm)
Creating service getstartedlab_visualizer (id: l9mnwkeq2jiononb5ihz9u7a4)
```

https://docs.docker.com/get-started/part5/

**Comment by Dzhuneyt** [comment]
@jony4 Have you tried the above command after actually updating some services? E.g. whenever I do changes to a service and push a new version of the image of a service (let's call the image `myprivaterepo.com/service-a:develop` for example), the above command doesn't get the service updated, because both its previous and new versions use the same tag: `myprivaterepo.com/service-a:develop`.

The reason I opened this issue is because I believe there should be an easier way to redeploy a stack, causing image digests to be resolved fresh from existing tags, defined in the docker-compose.yml file.

**Comment by thaJeztah** [comment]
Note that this repository is for Docker "classic" swarm, not for "swarm services" feature built in to the docker engine (which drives the docker stack deploy). docker stack deploy is a client-side feature that uses swarm services, and is maintained in the Docker cli (https://github.com/docker/cli).

I'll transfer this issue to that repository
